warehouse/query-core/src/main/java/datawave/query/util/GeoUtils.java

```diff
@@ -0,0 +1,475 @@
+     * 
+     * @param position
+     *            the numeric index position
+     * @return a bounding box geometry
+     */
+    public static Geometry positionToGeometry(long position) {
+        double[] latLon = positionToLatLon(position);
+        
+        double minLat = latLon[0];
+        double maxLat = minLat + 0.00001;
+        
+        double minLon = latLon[1];
+        double maxLon = minLon + 0.00001;
+        
+        Coordinate[] coords = new Coordinate[5];
+        coords[0] = new Coordinate(minLon, minLat);
+        coords[1] = new Coordinate(maxLon, minLat);
+        coords[2] = new Coordinate(maxLon, maxLat);
+        coords[3] = new Coordinate(minLon, maxLat);
+        coords[4] = new Coordinate(minLon, minLat);
+        
+        return gf.createPolygon(coords);
+    }
+    
+    /**
+     * Converts a geo index to its equivalent bounding box geometry
+     * 
+     * @param index
+     *            the geo index
+     * @return a bounding box geometry
+     */
+    public static Geometry indexToGeometry(String index) {
+        return positionToGeometry(indexToPosition(index));
+    }
+    
+    /**
+     * Converts a numeric index range to its equivalent geometry.
+     * 
+     * @param beginPosition
+     *            the minimum numeric index position
+     * @param endPosition
+     *            the maximum numeric index position
+     * @return a geometry which represents the range
+     */
+    public static Geometry positionRangeToGeometry(long beginPosition, long endPosition) {
+        List<long[]> ranges = decodePositionRange(beginPosition, endPosition);
+        
+        List<Geometry> geometries = new ArrayList<>();
+        for (long[] range : ranges) {
+            List<Double> latitudes = new ArrayList<>();
+            List<Double> longitudes = new ArrayList<>();
+            
+            Geometry minGeom = positionToGeometry(range[0]);
+            Arrays.stream(minGeom.getCoordinates()).forEach(c -> {
+                longitudes.add(c.x);
+                latitudes.add(c.y);
+            });
+            
+            double minLat = Collections.min(latitudes);
+            double minLon = Collections.min(longitudes);
+            latitudes.clear();
+            longitudes.clear();
+            
+            Geometry maxGeom = positionToGeometry(range[1] - 1);
+            Arrays.stream(maxGeom.getCoordinates()).forEach(c -> {
+                longitudes.add(c.x);
+                latitudes.add(c.y);
+            });
+            
+            double maxLat = Collections.max(latitudes);
+            double maxLon = Collections.max(longitudes);
+            
+            Coordinate[] coords = new Coordinate[5];
+            coords[0] = new Coordinate(minLon, minLat);
+            coords[1] = new Coordinate(maxLon, minLat);
+            coords[2] = new Coordinate(maxLon, maxLat);
+            coords[3] = new Coordinate(minLon, maxLat);
+            coords[4] = new Coordinate(minLon, minLat);
```

Comment:
save an object creation: coords[4] = coords[0];

Reply:
Easy enough. I'll make the change throughout.

Reply:
done